sources/web/datalab/jupyter.ts

```diff
@@ -426,10 +426,14 @@ function sendTemplate(key: string, data: common.Map<string>, response: http.Serv
 function responseHandler(proxyResponse: http.ClientResponse,
                          request: http.ServerRequest, response: http.ServerResponse) {
   if (proxyResponse.headers['access-control-allow-origin'] !== undefined) {
-    // Delete the allow-origin = * header that is sent (likely as a result of a workaround
-    // notebook configuration to allow server-side websocket connections that are
-    // interpreted by Jupyter as cross-domain).
-    delete proxyResponse.headers['access-control-allow-origin'];
+    if (appSettings.allowOriginOverride) {
```

Comment:
I wonder if we should set the header outside the "if (proxyResponse.headers['access-control-allow-origin'] !== undefined)" block? Not sure if Jupyter always sets the header but in case it doesn't, should we set the overriding header anyway?